File: core/src/com/inkostilation/pong/commands/QuitCommand.java
package com.inkostilation.pong.commands;

import com.inkostilation.pong.engine.IEngine;

import java.io.IOException;
import java.nio.channels.SocketChannel;

public class QuitCommand extends AbstractRequestCommand<IEngine<SocketChannel>, SocketChannel> {
    @Override
    public void execute(IEngine<SocketChannel> engine) {
        try {
            getEngine().quit(getMarker());
            getMarker().close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: core/src/com/inkostilation/pong/commands/QuitCommand.kt
package com.inkostilation.pong.commands

import com.inkostilation.pong.server.engine.AbstractEngine
import com.inkostilation.pong.server.engine.IEngine

class QuitCommand() : AbstractRequestCommand<AbstractEngine>() {
    override fun execute(input: AbstractEngine): Array<AbstractResponseCommand<*>> {
        return emptyArray()
    }

    override fun getEngineType() = AbstractEngine::class.java
}